buildSrc/src/main/kotlin/org/jetbrains/conventions/maven-cli-setup.gradle.kts

```diff
@@ -0,0 +1,62 @@
+package org.jetbrains.conventions
+
+import org.gradle.kotlin.dsl.support.serviceOf
+
+/**
+ * Utility for downloading and installing a Maven binary.
+ */
+
+plugins {
+    base
+}
+
+abstract class SetupMavenProperties {
+    val mavenVersion = "3.5.0"
+    val mavenPluginToolsVersion = "3.5.2"
```

Comment:
Very much optional, while we're at it: I think we actually forget to check/update the Maven version because it's buried in a place we don't visit that much. Extracting it to `gradle.properties` (or into the `toml` in the other PR) would not hurt

Reply:
Good point, I think so too. For now I'll put it in `gradle.properties`, and I think that later they can be moved to `libs.versions.toml` as part of #2884

Reply:
It was a little more work than I thought to refactor this, but I think it was worth it.

It revealed that the `CrossPlatformExec` task wasn't really needed, and could be replaced with a Provider that would determine the current operating system and use `mvn.cmd` on Windows, else use `mvn`.

Reply:
Oof, sorry for asking you to do that then, and thanks for doing! I do think it was worth it though!

Reply:
no apology needed :)